service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/DiagnosticReportMapper.java

```diff
@@ -73,16 +74,19 @@ public class DiagnosticReportMapper {
     private final ConfidentialityService confidentialityService;
 
     public String mapDiagnosticReportToCompoundStatement(DiagnosticReport diagnosticReport) {
-        List<Specimen> specimens = fetchSpecimens(diagnosticReport);
-        List<Observation> observations = fetchObservations(diagnosticReport);
+        List<Observation> initialObservations = fetchObservations(diagnosticReport);
+        List<Specimen> specimens = fetchSpecimens(diagnosticReport, initialObservations);
         final IdMapper idMapper = messageContext.getIdMapper();
-        markObservationsAsProcessed(idMapper, observations);
+        markObservationsAsProcessed(idMapper, initialObservations);
+
+        final List<Observation> processedObservations =
+                assignDummySpecimensToOrphanedTestResults(initialObservations, specimens);
 
         String mappedSpecimens = specimens.stream()
-            .map(specimen -> specimenMapper.mapSpecimenToCompoundStatement(specimen, observations, diagnosticReport))
+            .map(specimen -> specimenMapper.mapSpecimenToCompoundStatement(specimen, processedObservations, diagnosticReport))
```

Comment:
This set of `Observation` being passed in doesn't appear to have been filtered according to the specimen at all unless I'm mistaken.

Imagine we had 4 test results like below, how is the code as its currently written going to know which Observation matches to which Specimen?

```mermaid
graph TD;
    DiagnosticReport/1-->Specimen/1;
    DiagnosticReport/1-->Specimen/2;
    Specimen/1-->Observation/1;
    Specimen/1-->Observation/2;
    Specimen/2-->Observation/3;
    Specimen/2-->Observation/4;
```